download-basket/src/main/java/downloadbasket/helpers/DownloadServices.java

```diff
@@ -0,0 +1,277 @@
+import downloadbasket.data.LoadZipDetails;
+import org.apache.commons.mail.MultiPartEmail;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.activation.FileDataSource;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.util.ByteArrayDataSource;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.UUID;
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+
+/**
+ * Download services.
+ * @author Marko Kuosmanen
+ * @copyright Dimenteq Oy
+ *
+ */
+public class DownloadServices {
+	public static final String WFS_GETCAPABILITIES_XML_JSON_VALUE ="WFSGetCapabilitiesXML";
+	public static final String WFS_METADATA_URL_JSON_VALUE ="linkForMetadata";
+	public static final String WFS_USED_URL ="url";
+	public static final String WFS_FEATURETYPES ="featureTypes";
+	private final Logger LOGGER = LogFactory.getLogger(DownloadServices.class);
+
+	
+	/**
+	 * Default Constructor.
+	 */
+	public DownloadServices(){
+	}
+	
+	/**
+	 * Enum of Http methods.
+	 * @author Marko Kuosmanen
+	 *
+	 */
+	public enum HttpMetodes {
+		   POST("POST"), GET("GET");
+		   private final String stringValue;
+		   private HttpMetodes(final String s) { stringValue = s; }
+		   public String toString() { return stringValue; }
+	}
+	
+	/**
+	 * Load shape-ZIP from Geoserver.
+	 * @param ldz load zip details
+	 * @return filename file name
+	 * @throws IOException 
+	 */
+	public String loadZip(LoadZipDetails ldz) throws IOException{
+		String realFileName = "";
+		String returnFileName = "";
+		OutputStreamWriter writer = null;
+		HttpURLConnection conn = null;
+		InputStream istream = null;
+		OutputStream ostream = null;
+		
+		try{
+			System.out.println(ldz.getWFSUrl());
+			
+			
+			if(ldz.isDownloadNormalWay()) {
+				System.out.println("Download normal way");
+			} else {
+				System.out.println("Download plugin way");
+			}
+			
+			if(ldz.getGetFeatureInfoRequest().isEmpty()){
+				return null;
+			}
+			
+			LOGGER.debug("WFS URL: " + ldz.getWFSUrl());
+			LOGGER.debug("-- filtter: " + ldz.getGetFeatureInfoRequest());
+			final URL url = new URL(ldz.getWFSUrl() + ldz.getGetFeatureInfoRequest());
+			
+			conn = (HttpURLConnection) url.openConnection();
+			conn.setRequestMethod("GET");
+			conn.setConnectTimeout(600000);
+
+			conn.connect();
+		
+			String filename = UUID.randomUUID().toString();
+			istream = conn.getInputStream();
+			
+			String strTempDir = ldz.getTemporaryDirectory();
+			
+			File dir0 = new File(strTempDir);
+			dir0.mkdirs();
+			
+			String slashType = "";
+			String myFullFileName = dir0.getName();
+			if (myFullFileName.lastIndexOf("\\")> 0) {
+				slashType = "\\";
+			} else {
+				slashType = "/";
+			}
+			
+			realFileName = strTempDir + slashType + filename + ".zip";
+			
+			returnFileName = strTempDir + slashType + filename + ".zip";
+			ostream = new FileOutputStream(realFileName);
```

Comment:
Use File myFile = new File(dir0, filename + ".zip"); instead so you can remove the slashType checking.